evalml/pipelines/components/transformers/preprocessing/datetime_featurizer.py

```diff
@@ -91,19 +87,19 @@ def transform(self, X, y=None):
             ww.DataTable: Transformed X
         """
         X_ww = infer_feature_types(X)
-        X_t = _convert_woodwork_types_wrapper(X_ww.to_dataframe())
         features_to_extract = self.parameters["features_to_extract"]
         if len(features_to_extract) == 0:
-            return infer_feature_types(X_t)
+            return X_ww
         for col_name in self._date_time_col_names:
             for feature in features_to_extract:
                 name = f"{col_name}_{feature}"
-                features, categories = self._function_mappings[feature](X_t[col_name], self.encode_as_categories)
-                X_t[name] = features
+                features, categories = self._function_mappings[feature](X_ww[col_name], self.encode_as_categories)
+                features.name = None
+                X_ww.ww[name] = features
                 if categories:
                     self._categories[name] = categories
-        X_t = X_t.drop(self._date_time_col_names, axis=1)
-        return _retain_custom_types_and_initalize_woodwork(X_ww, X_t)
+        X_t = X_ww.ww.drop(self._date_time_col_names)
+        return X_t
```

Comment:
No need to call `_retain_custom_types_and_initialize_woodwork` here because our implementation preserves the schema.